pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -37,26 +38,63 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, ci *CloudInfo)
 	return allErrs
 }
 
-// validateMachinesSubnet validates the machines subnet and enforces proper byo subnet usage and returns a list of all validation errors
-func validateMachinesSubnet(p *openstack.Platform, n *types.Networking, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
-	if p.MachinesSubnet != "" {
-		if len(p.ExternalDNS) > 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, "externalDNS is set, externalDNS is not supported when machinesSubnet is set"))
-		}
-		if ci.MachinesSubnet == nil {
-			allErrs = append(allErrs, field.NotFound(fldPath.Child("machinesSubnet"), p.MachinesSubnet))
-		} else if !validUUIDv4(p.MachinesSubnet) {
-			allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), errors.New("invalid subnet ID")))
-		} else {
-			if n.MachineNetwork[0].CIDR.String() != ci.MachinesSubnet.CIDR {
-				allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), fmt.Errorf("the first CIDR in machineNetwork, %s, doesn't match the CIDR of the machineSubnet, %s", n.MachineNetwork[0].CIDR.String(), ci.MachinesSubnet.CIDR)))
+// validateControlPlanePort validates the machines subnets and network, while enforcing proper byo subnets usage and returns a list of all validation errors.
+func validateControlPlanePort(p *openstack.Platform, n *types.Networking, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
+	networkID := ""
+	if p.ControlPlanePort != nil {
+		if len(p.ControlPlanePort.FixedIPs) <= 2 {
+			if len(p.ExternalDNS) > 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, "externalDNS is set, externalDNS is not supported when ControlPlanePort is set"))
+			}
+			for _, fixedIP := range p.ControlPlanePort.FixedIPs {
```

Comment:
Feels like we should also validate if the subnets are IPv4 and IPv6. I think it's possible to have two IPv4 addresses in a port and I don't think we would want to support that?